handlers/graph.go

```diff
@@ -280,23 +301,39 @@ func populateTrafficMap(trafficMap graph.TrafficMap, vector *model.Vector, o opt
 			handleMisconfiguredLabels(dest, destApp, destVer, val, o)
 		}
 
-		var ck string
-		switch {
-		case strings.HasPrefix(string(code), "2"):
-			ck = "rate2xx"
-		case strings.HasPrefix(string(code), "3"):
-			ck = "rate3xx"
-		case strings.HasPrefix(string(code), "4"):
-			ck = "rate4xx"
-		case strings.HasPrefix(string(code), "5"):
-			ck = "rate5xx"
-		}
-		addToRate(edge.Metadata, ck, val)
-		addToRate(edge.Metadata, "rate", val)
+		if len(tcpMetric) == 0 {
+			var ck string
+			switch {
+			case strings.HasPrefix(string(code), "2"):
+				ck = "rate2xx"
+			case strings.HasPrefix(string(code), "3"):
+				ck = "rate3xx"
+			case strings.HasPrefix(string(code), "4"):
+				ck = "rate4xx"
+			case strings.HasPrefix(string(code), "5"):
+				ck = "rate5xx"
+			}
+			addToRate(edge.Metadata, ck, val)
+			addToRate(edge.Metadata, "rate", val)
 
-		addToRate(source.Metadata, "rateOut", val)
-		addToRate(dest.Metadata, ck, val)
-		addToRate(dest.Metadata, "rate", val)
+			addToRate(source.Metadata, "rateOut", val)
+			addToRate(dest.Metadata, ck, val)
+			addToRate(dest.Metadata, "rate", val)
+		} else {
+			source.Metadata["hasTcpTraffic"] = true
```

Comment:
Why is it "hasTcpTraffic" in the metadata but the flag is "hasTcp"?  Maybe just use hasTcp everywhere.  Do we actually need the hasTcp flag in the json?  If so, do we also now need a hasHttp?  If we can determine the value of these flags by looking at the rate information perhas we could just set these convenience flags on the ui-side.

Reply:
Ok, I removed the flag. Yes... it can be determined by looking at the rate data.